packages/lib/test/mocks/mock-solc-0.6/contracts/initializable/MultipleInheritanceInitializableMocks.sol

```diff
@@ -0,0 +1,79 @@
+pragma solidity ^0.6.0;
+
+import "./Initializable.sol";
+
+// Sample contracts showing upgradeability with multiple inheritance.
+// Child contract inherits from Father and Mother contracts, and Father extends from Gramps.
+// 
+//         Human
+//       /       \
+//      |       Gramps
+//      |         |
+//    Mother    Father
+//      |         |
+//      -- Child --
+
+/**
+ * Sample base intializable contract that is a human
+ */
+contract SampleHuman is Initializable {
+  bool public isHuman;
+
+  function initialize() initializer virtual public {
+    isHuman = true;
+  }
+}
+
+/**
+ * Sample base intializable contract that defines a field mother
+ */
+contract SampleMother is Initializable, SampleHuman {
+  uint256 public mother;
+
+  function initialize(uint256 value) initializer virtual public {
+    SampleHuman.initialize();
+    mother = value;
+  }
+}
+
+/**
+ * Sample base intializable contract that defines a field gramps
+ */
+contract SampleGramps is Initializable, SampleHuman {
+  uint256 public gramps;
+
+  function initialize(uint256 value) initializer virtual public {
+    SampleHuman.initialize();
+    gramps = value;
+  }
+}
+
+/**
+ * Sample base intializable contract that defines a field father and extends from gramps
+ */
+contract SampleFather is Initializable, SampleGramps {
+  uint256 public father;
+
+  function initialize(uint256 _gramps, uint256 _father) initializer virtual public {
+    SampleGramps.initialize(_gramps);
+    father = _father;
+  }
+}
+
+/**
+ * Child extends from mother, father (gramps)
+ */
+contract SampleChild is Initializable, SampleMother, SampleFather {
+  uint256 public child;
+
+  function initialize(uint256 value) initializer override(SampleGramps, SampleMother) public {
+    SampleGramps.initialize(value);
+    SampleMother.initialize(value);
+  }
+
+  function initialize(uint256 _mother, uint256 _gramps, uint256 _father, uint256 _child) initializer public {
+    SampleMother.initialize(_mother);
+    SampleFather.initialize(_gramps, _father);
+    child = _child;
+  }
+}
```

Comment:
Again, couldn't run the code so my question is why this doesn't cause to a duplicated call at `SampleHuman. initialize`? Since both `SampleMother` and `SampleFather` are children of `SampleHuman`.

Reply:
Actually, it does, and it's one of the things tested here (note that this test existed from before, I just moved it and migrated it to 0.6). There was a bug in Initializable that caused it to fail in these scenarios with an _already initialized_ error. This test checks that it's ok to call a base initializer more than once during the initialization process, to accommodate for structures like this one.

Reply:
Hmm. Didn't know that our current approach is that your `init` method might be called twice. This is certainly tricky and not many devs would expect that. I like the direction we are moving with Transpiler.